src/main/java/core/mate/academy/service/MachineServiceImpl.java

```diff
@@ -1,7 +1,47 @@
 package core.mate.academy.service;
 
+import core.mate.academy.model.Bulldozer;
+import core.mate.academy.model.Excavator;
+import core.mate.academy.model.Machine;
+import core.mate.academy.model.Truck;
+import java.util.ArrayList;
+import java.util.List;
+
 /**
  * Your implementation of MachineService.
  */
-public class MachineServiceImpl {
+public class MachineServiceImpl implements MachineService<Machine> {
+    @Override
+    public List<Machine> getAll(Class<? extends Machine> type) {
+        MachineProducer bulldozers = new BulldozerProducer();
+        MachineProducer trucks = new TruckProducer();
+        MachineProducer excavators = new ExcavatorProducer();
```

Comment:
Put these inside if body for memory and readability improvement. I mean if type is Bulldozer, why create two other?